src/components/Card/Card.tsx

```diff
@@ -48,44 +47,8 @@ function Card({
     };
   }, [cardholderRef]);
 
-  const [springValues, springRef] = useSpring(
-    () => ({
-      x: 0,
-      y: 0,
-      config: {
-        mass: 2,
-        friction: 14,
-        tension: 120,
-      },
-    }),
-    []
-  );
-
-  const handleMouseLeave = () => {
-    if (cardholderRef.current) {
-      const regex =
-        /translate\((?<offsetX>[-\d.]+)px,\s*(?<offsetY>[-\d.]+)px\)/;
-      const transformValue = cardholderRef.current.style.transform;
-      const matches = regex.exec(transformValue);
-
-      if (matches?.groups) {
-        const { offsetX, offsetY } = matches.groups;
-        springRef.start({
-          from: {
-            x: parseFloat(offsetX),
-            y: parseFloat(offsetY),
-          },
-          to: {
-            x: 0,
-            y: 0,
-          },
-        });
-      }
-    }
-  };
-  const handleMouseEnter = () => {
-    springRef.stop();
-  };
+  const { springValues, handleMouseEnter, handleMouseLeave } =
+    useSpringAnimation(cardholderRef);
 
   return (
     <animated.div
```

Comment:
Take a look at these two very similar libs for conditional class names:
- https://github.com/lukeed/clsx#usage
- https://github.com/JedWatson/classnames#usage

They allow to replace this:
```ts
`${styles.card} ${flip ? styles.flipped : ""} 
${off ? `${styles.off} ${styles.flipped}` : "" }`
```
with something like:
```js
classNames({
  [styles.card]: true,
  [styles.flipped]: flip || off,
  [styles.off]: off,
})
```

To me it look more clear. Bu tit's up to you